scripts/zones/Port_Jeuno/npcs/Squintrox_Dryeyes.lua

```diff
@@ -3,117 +3,534 @@
-            sLuna and
-            sAstrum and
-            count == 3 and
-            acpMission >= xi.mission.id.acp.GATHERER_OF_LIGHT_I and
-            not hasCrimsonKey and
-            os.time() > lastCrimson
-        then -- and timer stuff here) then
+            not player:hasKeyItem(ki) and
+            now >= player:getCharVar('LastCrimsonKey')
+        then
             player:tradeComplete()
-            player:addKeyItem(xi.ki.CRIMSON_KEY)
-            player:setCharVar('LastCrimsonKey', getMidnight())
+            player:addKeyItem(ki)
+            player:setCharVar('LastCrimsonKey', nextMidnight)
             player:messageSpecial(ID.text.DRYEYES_2)
-            player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.CRIMSON_KEY)
-
-        elseif
-            sLux and
-            sLuna and
-            sAstrum and
-            count == 3 and
-            (os.time() <= lastCrimson or hasCrimsonKey)
+            player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+        else
+            player:messageSpecial(ID.text.DRYEYES_3, ki)
+        end
+    -- White Coral Key: trade 3 drops - orcish plate armor, quadav backscale, yagudo caulk
+    elseif
+        hasWhiteCoralItems and
+        finishedAMK
+    then
+        local ki = xi.ki.WHITE_CORAL_KEY
+        if
+            not player:hasKeyItem(ki) and
+            now >= player:getCharVar('LastWhiteCoralKey')
         then
-            player:messageSpecial(ID.text.DRYEYES_3, xi.ki.CRIMSON_KEY)
-        -- White Coral Key:
-        -- elseif xi.settings.main.ENABLE_AMK == 1 and
-            -- haven't even started AMK related trades yet.
+            player:tradeComplete()
+            player:addKeyItem(ki)
+            player:setCharVar('LastWhiteCoralKey', nextMidnight)
+            player:messageSpecial(ID.text.DRYEYES_2)
+            player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+        else
+            player:messageSpecial(ID.text.DRYEYES_3, ki)
         end
+    -- Moogle Key: trade proper enfeebling kit
+    elseif
+        hasEnfeebKit and
+        finishedASA
+    then
+        local ki = xi.ki.MOOGLE_KEY
+        if
+            not player:hasKeyItem(ki) and
+            now >= player:getCharVar('LastMoogleKey')
+        then
+            player:tradeComplete()
+            player:addKeyItem(ki)
+            player:setCharVar('LastMoogleKey', nextMidnight)
+            player:setCharVar('ASA_kit', 0)
+            player:messageSpecial(ID.text.DRYEYES_2)
+            player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+        else
+            player:messageSpecial(ID.text.DRYEYES_3, ki)
+        end
+    else
+        player:showText(npc, ID.text.GET_LOST)
     end
 end
 
 entity.onTrigger = function(player, npc)
-    if
-        xi.settings.main.ENABLE_ACP == 0 and
-        xi.settings.main.ENABLE_AMK == 0 and
-        xi.settings.main.ENABLE_ASA == 0
-    then
+    local now          = os.time()
+    local nextMidnight = getMidnight()
```

Comment:
These two aren't used in here

Reply:
Good call I copy pasted definitions for consistent naming